File: src/main/java/com/github/simkuenzi/restforms/MandatoryField.java
package com.github.simkuenzi.restforms;

import org.jetbrains.annotations.NotNull;

public class MandatoryField implements FormField<String> {
    private final FormField<String> inner;
    private final String message;


    public MandatoryField(FormField<String> inner, String message) {
        this.inner = inner;
        this.message = message;
    }

    @Override
    public String getValue() {
        return inner.getValue();
    }

    @Override
    public boolean isValid() {
        return !getValue().isEmpty();
    }

    @NotNull
    @Override
    public String getMessage() {
        return !isValid() ? message : inner.getMessage();
    }

    @Override
    public String toString() {
        return inner.toString();
    }
}

File: src/main/kotlin/com/github/simkuenzi/restforms/MandatoryField.kt
package com.github.simkuenzi.restforms

class MandatoryField(private val inner: FormField<String>, private val message: String) : FormField<String> {
    override fun name() = inner.name()
    override fun rawValue() = inner.rawValue()
    override fun value() = inner.value()
    override fun valid() = !value().isEmpty()
    override fun message() = if (!valid()) message else inner.message()
    override fun toString() = inner.toString()
}
